repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -0,0 +1,53 @@
+package ammonite.repl
+
+import acyclic.file
+import java.io.{File, FileInputStream, IOException, FileWriter}
+import org.yaml.snakeyaml.Yaml
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.JavaConversions.asScalaBuffer
+
+object Storage{
+  val dir = new java.io.File(System.getProperty("user.home") + "/.ammonite")    
+  if(dir.exists){
+    if(!dir.isDirectory){
+      dir.delete()
+      dir.mkdir()
+    }
+  } else {
+    dir.mkdir()
+  }
+
+  lazy val yaml = new Yaml
+
+  def loadHistory: History = {
+    try{
+      val res = new History
+      val list = yaml.synchronized{
+        yaml.load(new FileInputStream(dir + "/history"))
+      }
+      list match {
+        case a: java.util.List[String] => res ++= a
+        case _ =>
+      }
+      res
+    } catch {
+      case e: IOException => new History
+    }
+  }
+
+  def saveHistory(h: History): Unit = {
```

Comment:
This one seems to be mostly linear wrt file size.I tested with short lines, 1000 lines gave 1.3ms, 10000 lines:  14 ms 100000 lines: 168ms. I benchmarked the whole `saveHistory` function so YAML serialization is included. I'd say it is acceptable but we will need to make it append to the file some time.